src/common/src/util/value_encoding/mod.rs

```diff
@@ -31,6 +35,190 @@ use error::ValueEncodingError;
 
 pub type Result<T> = std::result::Result<T, ValueEncodingError>;
 
+#[derive(Clone, Copy)]
+enum Width {
```

Comment:
Please move all of these to a new module under `value_encoding` crate, hide those interfaces that should not be public, and add rustdoc for all public stuff.